attention-codelab/process_figshare.py

```diff
@@ -0,0 +1,107 @@
+"""
+
+Cleans and splits the toxicity data from Figshare: 
+
+https://figshare.com/articles/Wikipedia_Talk_Labels_Toxicity/4563973
+
+------------------------------------------------------------------------
+
+Copyright 2018, Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import pandas as pd
+import os
+from urllib.request import urlretrieve
+
+DEFAULT_DATA_DIR = 'data/'
+
+def download_figshare(download_data_dir = DEFAULT_DATA_DIR):
+    """
+    Downloads the toxicity data from Figshare.
+    
+    Args:
+          * download_data_dir (string): if provided, the directory where the Figshare tsvs should be stored
+    """
+    if not os.path.exists(download_data_dir):
+        os.makedirs(download_data_dir)
+    
+    already_exist = True
+    for file in ['toxicity_annotations.tsv', 'toxicity_annotated_comments.tsv']:
+        if not os.path.isfile(os.path.join(download_data_dir, file)):
+            already_exist = False
+    
+    if already_exist:
+        print('Figshare data already exists.')
+        return
+    
+    print('Downloading toxicity_annotations.tsv...', end = '')
+    urlretrieve('https://ndownloader.figshare.com/files/7394539', 
+                os.path.join(download_data_dir, 'toxicity_annotations.tsv'))
```

Comment:
why don't you do this in the loop above
if not os.path.isfile(...):
  urlretreive(..., os.path.join(file))
also pull ndownload.figshare.com prefix to a constant string somewhere.

Reply:
Good suggestion! Done.